src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,216 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/tree.css';
+import cx from 'classnames';
+
+export const TreeContext = React.createContext<
+  | {
+      nodeDepth: number;
+      subNodeIds?: string[];
+      hasSubNodes: boolean;
+    }
+  | undefined
+>(undefined);
+
+export type NodeData<T> = {
+  subNodes?: Array<T>;
+  nodeId: string;
+  node: T;
+  isExpanded?: boolean;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  hasSubNodes: boolean;
+};
+
+type FlatNode<T> = {
+  nodeData: NodeData<T>;
+  depth: number;
+  subNodeIds?: Array<string>;
+};
+
+export type TreeProps<T> = {
+  /**
+   * Custom renderer for tree nodes, recommended to use TreeNode component.
```

Comment:
If you wrap `TreeNode` with ` would be better readability

Reply:
Done